ddl/ddl_api.go

```diff
@@ -5382,11 +5390,17 @@ func (d *ddl) AlterTableTTLInfoOrEnable(ctx sessionctx.Context, ident ast.Ident,
 			TableName:  tableName,
 			Type:       model.ActionAlterTTLInfo,
 			BinlogInfo: &model.HistoryInfo{},
-			Args:       []interface{}{ttlInfo, ttlEnable},
+			Args:       []interface{}{ttlInfo, ttlEnable, ttlCronJobSchedule},
 		}
 	} else {
 		if tblInfo.TTLInfo == nil {
-			return errors.Trace(dbterror.ErrSetTTLEnableForNonTTLTable)
+			if ttlEnable != nil && ttlCronJobSchedule == nil {
+				return errors.Trace(dbterror.ErrSetTTLOptionForNonTTLTable.FastGenByArgs("TTL_ENABLE"))
+			} else if ttlEnable == nil && ttlCronJobSchedule != nil {
+				return errors.Trace(dbterror.ErrSetTTLOptionForNonTTLTable.FastGenByArgs("TTL_CRONJOB_SCHEDULE"))
+			} else {
+				return errors.Trace(dbterror.ErrSetTTLOptionForNonTTLTable.FastGenByArgs("TTL_ENABLE and TTL_CRONJOB_SCHEDULE"))
+			}
 		}
 
 		job = &model.Job{
```

Comment:
Seems we can move the `job = &model.Job{...` outside the block to reduce some dup codes.

Reply:
Done.